src/main/scala/ai/privado/languageEngine/ruby/passes/download/ExternalDependenciesPass.scala

```diff
@@ -0,0 +1,82 @@
+package ai.privado.languageEngine.ruby.passes.download
+
+import ai.privado.languageEngine.ruby.cache.PackageTable
+import ai.privado.languageEngine.ruby.download.JRubyBasedParser
+import better.files.File
+import io.shiftleft.codepropertygraph.Cpg
+import io.shiftleft.passes.ConcurrentWriterCpgPass
+import org.jruby.Ruby
+import org.jruby.ast.{Colon2Node, DefnNode, Node, NodeType, ReturnNode}
+
+import scala.collection.mutable.ListBuffer
+import scala.jdk.CollectionConverters.CollectionHasAsScala
+
+class ExternalDependenciesPass(cpg: Cpg, tempExtDir: String, packageTable: PackageTable, inputPath: String)
+    extends ConcurrentWriterCpgPass[String](cpg) {
+  private val rubyInstance = Ruby.getGlobalRuntime()
+
+  override def generateParts(): Array[String] =
+    getRubyDependenciesFile(inputPath) ++ getRubyDependenciesFile(tempExtDir)
+
+  override def runOnPart(diffGraph: DiffGraphBuilder, filePath: String): Unit = {
+    val moduleName = resolveModuleNameFromPath(filePath)
+    try {
+      processRubyDependencyFile(filePath, moduleName)
+    } catch {
+      case ex: Exception =>
+        println(s"Error while parsing $moduleName module File ($filePath): ${ex.getMessage}")
+    }
+  }
+
+  private def fetchMethodInfoFromNode(node: Node, currentNameSpace: ListBuffer[String], moduleName: String): Unit = {
+    if (node != null) {
+      node.getNodeType match {
+        case NodeType.CLASSNODE | NodeType.MODULENODE =>
+          val childList = node.childNodes().asScala.toList
+          if (childList.nonEmpty) {
+            val classOrModuleName = childList.head.asInstanceOf[Colon2Node].getName.toString
+            currentNameSpace.addOne(classOrModuleName)
+          }
+        case NodeType.DEFNNODE =>
+          val methodName = node.asInstanceOf[DefnNode].getName.toString
+          val classPath  = currentNameSpace.mkString(":")
+          packageTable.addPackageMethod(moduleName, methodName, classPath, "<extMod>")
+        case _ =>
+      }
+      node.childNodes().forEach(childNode => fetchMethodInfoFromNode(childNode, currentNameSpace, moduleName))
+    }
+  }
+
+  private def processRubyDependencyFile(inputPath: String, moduleName: String): Unit = {
+    val currentFile = File(inputPath)
+    if (currentFile.exists) {
```

Comment:
Can we not check this directly like `File(inputPath).exists`?  Instead of creating a variable